src/NuGet.Core/NuGet.Protocol/Model/LocalPackageSearchMetadata.cs

```diff
@@ -145,5 +147,28 @@ public string LoadFileAsText(string path)
             }
             return fileContent;
         }
+
+        /// <summary>
+        /// Points to an Icon, either Embedded Icon or IconUrl
+        /// </summary>
+        public Uri GetIcon()
```

Comment:
Does this need to be public given that it's exposed by the `IconUrl` property?

Reply:
It is better to make the method private. If it is needed to be public. We can address it later.